gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/GradleUtil.java

```diff
@@ -95,6 +98,33 @@ public static JavaProject convertGradleProject(Project gradleProject) {
         .build();
   }
 
+  public static List<Plugin> getPluginWithKubernetesExtensionConfigured(List<Plugin> plugins, List<String> legacyPlugins, KubernetesExtension kubernetesExtension) {
```

Comment:
How is this method related with Gradle?

Is it possible that this method is more suitable in the KubernetesExtension class or elsewhere?

Considering that this "getter" is used from `AbstractJKubeTask`: 
```java
kubernetesExtension.javaProject.setPlugins(GradleUtil.getPluginWithKubernetesExtensionConfigured(kubernetesExtension.javaProject.getPlugins(), kubernetesExtension.javaProject.getGradlePlugins(), kubernetesExtension));
```

this looks like a really big roundtrip.

It's also overriding the plugins that were initially set in L81 of this same class.

I will try to check this locally because it's really smelly.